tracer/src/Datadog.Trace/Vendors/ICSharpCode.SharpZipLib/Encryption/ZipAESTransform.cs

```diff
@@ -0,0 +1,229 @@
+using Datadog.Trace.Vendors.ICSharpCode.SharpZipLib.Core;
+
+namespace Datadog.Trace.Vendors.ICSharpCode.SharpZipLib.Encryption
+{
+	/// <summary>
+	/// Transforms stream using AES in CTR mode
+	/// </summary>
+	internal class ZipAESTransform : ICryptoTransform
+	{
+#if NETFRAMEWORK
+		class IncrementalHash : HMACSHA1
+		{
+			bool _finalised;
+			public IncrementalHash(byte[] key) : base(key) { }
+			public static IncrementalHash CreateHMAC(string n, byte[] key) => new IncrementalHash(key);
+			public void AppendData(byte[] buffer, int offset, int count) => TransformBlock(buffer, offset, count, buffer, offset);
+			public byte[] GetHashAndReset()
+			{
+				if (!_finalised)
+				{
+					byte[] dummy = new byte[0];
+					TransformFinalBlock(dummy, 0, 0);
+					_finalised = true;
+				}
+				return Hash;
+			}
+		}
+
+		static class HashAlgorithmName
+		{
+			public static string SHA1 = null;
+		}
+#endif
+
+		private const int PWD_VER_LENGTH = 2;
+
+		// WinZip use iteration count of 1000 for PBKDF2 key generation
+		private const int KEY_ROUNDS = 1000;
+
+		// For 128-bit AES (16 bytes) the encryption is implemented as expected.
+		// For 256-bit AES (32 bytes) WinZip do full 256 bit AES of the nonce to create the encryption
+		// block but use only the first 16 bytes of it, and discard the second half.
+		private const int ENCRYPT_BLOCK = 16;
+
+		private int _blockSize;
+		private readonly ICryptoTransform _encryptor;
+		private readonly byte[] _counterNonce;
+		private byte[] _encryptBuffer;
+		private int _encrPos;
+		private byte[] _pwdVerifier;
+		private IncrementalHash _hmacsha1;
+		private byte[] _authCode = null;
+
+		private bool _writeMode;
+
+		/// <summary>
+		/// Constructor.
+		/// </summary>
+		/// <param name="key">Password string</param>
+		/// <param name="saltBytes">Random bytes, length depends on encryption strength.
+		/// 128 bits = 8 bytes, 192 bits = 12 bytes, 256 bits = 16 bytes.</param>
+		/// <param name="blockSize">The encryption strength, in bytes eg 16 for 128 bits.</param>
+		/// <param name="writeMode">True when creating a zip, false when reading. For the AuthCode.</param>
+		///
+		public ZipAESTransform(string key, byte[] saltBytes, int blockSize, bool writeMode)
+		{
+			if (blockSize != 16 && blockSize != 32) // 24 valid for AES but not supported by Winzip
+				throw new Exception("Invalid blocksize " + blockSize + ". Must be 16 or 32.");
+			if (saltBytes.Length != blockSize / 2)
+				throw new Exception("Invalid salt len. Must be " + blockSize / 2 + " for blocksize " + blockSize);
+			// initialise the encryption buffer and buffer pos
+			_blockSize = blockSize;
+			_encryptBuffer = new byte[_blockSize];
+			_encrPos = ENCRYPT_BLOCK;
+
+			// Performs the equivalent of derive_key in Dr Brian Gladman's pwd2key.c
+			var pdb = new Rfc2898DeriveBytes(key, saltBytes, KEY_ROUNDS);
+			var rm = Aes.Create();
+			rm.Mode = CipherMode.ECB;           // No feedback from cipher for CTR mode
```

Comment:
@zacharycmontoya what do you want to do about this? Can/should we strip out this from the vendored code? We don't need encryption right, so this is just adding bloat anyway? 🤔

Reply:
I was thinking about just dismissing the alert instead of stripping out the vendored code. Otherwise, if that doesn't seem wise, I can just manually update the `ZipAESTransform` constructor to throw, that way we will fail loudly if we ever try to do encryption. Do you have a preference? `ZipFile`, `ZipOutputStream`, and `ZipAESStream` all take a dependency on the class so instead of pruning it out meticulously, updating only one place by throwing an exception for the object constructor seems like the best solution to me